crypto/evp/evp_fetch.c

```diff
@@ -139,14 +162,17 @@ void *evp_generic_fetch(OPENSSL_CTX *libctx, int operation_id,
 {
     OSSL_METHOD_STORE *store = get_default_method_store(libctx);
     OSSL_NAMEMAP *namemap = ossl_namemap_stored(libctx);
-    int id;
+    int nameid;
+    uint32_t methid;
     void *method = NULL;
 
     if (store == NULL || namemap == NULL)
         return NULL;
 
-    if ((id = ossl_namemap_number(namemap, name)) == 0
-        || !ossl_method_store_cache_get(store, id, properties, &method)) {
+    if ((nameid = ossl_namemap_number(namemap, name)) == 0
+        || !ossl_method_store_cache_get(store,
+                                        method_id(operation_id, nameid),
```

Comment:
method_id could fail

Reply:
Doesn't matter much, 'cause we know that there will never be a method with calculated id being zero in the cache.  So if `method_id` fails, so will `ossl_method_store_cache_get`

Reply:
ok

Reply:
I added a safety check anyway...